taichi/backends/vulkan/codegen_vulkan.cpp

```diff
@@ -1224,16 +1245,19 @@ class TaskCodegen : public IRVisitor {
     std::vector<BufferBind> result;
     int binding = 0;
 
-    auto bind_buffer = [&](TaskAttributes::Buffers buffer) {
+    auto bind_buffer = [&](BufferInfo buffer) {
       result.push_back({buffer, binding});
       buffer_binding_map_[buffer] = binding++;
     };
 
-    bind_buffer(BuffersEnum::Root);
-    bind_buffer(BuffersEnum::GlobalTmps);
+    for (int root = 0; root < compiled_structs_.size(); ++root) {
```

Comment:
BTW, if a kernel only uses the second root, will this still work? (I.e. do we still need to bind the first root?)

Reply:
I think it's fine, as the descriptor layout is locked.

Reply:
This will work, but there will indeed be an extra unused binding in that case.